src/Commands/ObserverMakeCommand.php

```diff
@@ -0,0 +1,171 @@
+{
+    use ModuleCommandTrait;
+
+    /**
+     * The console command name.
+     *
+     * @var string
+     */
+    protected $name = 'module:make-observer';
+
+    protected $argumentName = 'name';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Create a new observer class for the specified module.';
+
+    /**
+     * Get the default namespace for the class.
+     *
+     * @param  string  $rootNamespace
+     * @return string
+     */
+    public function getDefaultNamespace() : string
+    {
+        $module = $this->laravel['modules'];
+
+        return $module->config('paths.generator.observers.namespace') ?: $module->config('paths.generator.observers.path', 'Observers');
+    }
+
+    /**
+     * Get the default namespace for the class.
+     *
+     * @param  string  $rootNamespace
+     * @return string
+     */
+    public function getModelNamespace() : string
+    {
+        $module = $this->laravel['modules'];
+
+        return $module->config('paths.generator.model.namespace') ?: $module->config('paths.generator.model.path', 'Entities');
+    }
+
+    /**
+     * Get template contents.
+     *
+     * @return string
+     */
+    protected function getTemplateContents()
+    {
+        $module = $this->laravel['modules']->findOrFail($this->getModuleName());
+
+        if($this->option('model') ) {
+            return (new Stub('/observer.stub', [
+                'NAMESPACE' => $this->getClassNamespace($module),
+                'CLASS'     => $this->getClass(),
+
+                'NAMESPACEDMODEL'     => $this->getNamespacedModel(),
+                'DUMMYMODEL'     => $this->getModel(),
+                'MODEL'     => $this->getModelVariable(),
+            ]))->render();
+        }
+
+        return (new Stub('/observer.plain.stub', [
+            'NAMESPACE' => $this->getClassNamespace($module),
+            'CLASS'     => $this->getClass(),
+        ]))->render();
+    }
+
+    /**
+     * Get the destination file path.
+     *
+     * @return string
+     */
+    protected function getDestinationFilePath()
+    {
+        $path = $this->laravel['modules']->getModulePath($this->getModuleName());
+
+        $notificationPath = GenerateConfigReader::read('observers');
+
+        return $path . $notificationPath->getPath() . '/' . $this->getFileName() . '.php';
+    }
+
+    /**
+     * @return string
+     */
+    private function getFileName()
+    {
+        return Str::studly($this->argument('name'));
+    }
+
+    /**
+     * @return string
+     */
+    private function getNamespacedModel()
+    {
+        $model = $this->option('model');
+
+        $model = str_replace('/', '\\', $model);
+
+        $module = $this->laravel['modules']->findOrFail($this->getModuleName());
+        $namespaceModel = $this->getClassNamespace($module, $this->getModelNamespace() ) . '\\' . $model;
+
+        if (Str::startsWith($model, '\\')) {
```

Comment:
Why is all this needed?

Reply:
Generate namespace for model, of cource. Do you know a better way?

Reply:
I got that part thanks. I'm asking why replacing slashes.

Reply:
I will take a look and say later. But without this, it didn’t work.